app/src/main/java/com/x8bit/bitwarden/ui/platform/base/util/EventsEffect.kt

```diff
@@ -12,10 +15,15 @@ import kotlinx.coroutines.flow.onEach
 @Composable
 fun <E> EventsEffect(
     viewModel: BaseViewModel<*, E, *>,
+    lifecycleOwner: Lifecycle = LocalLifecycleOwner.current.lifecycle,
     handler: suspend (E) -> Unit,
 ) {
     LaunchedEffect(key1 = Unit) {
         viewModel.eventFlow
+            .filter {
+                it is BackgroundEvent ||
```

Comment:
Can we document at the function-level the restrictions on observing events? Something like:

```kotlin
/**
 * Convenience method for observing event flow from [BaseViewModel].
 * 
 * By default, events will only be consumed when the associated screen is 
 * resumed to avoid things like duplicate navigation calls. In order to override
 * this behavior, a given event type can implement `BackgroundEvent`.
 */
@Composable
fun <E> EventsEffect(
```